### PR TITLE
Add tx signing metrics

### PR DESCRIPTION
**Pull request by t-bast**
Monitor the rate at which we sign channel txs and the duration of the signing operations.

### CODE REVIEW

**Review by pm47 — COMMENTED** [review]
The PR is fine, but should we rename the `sign` methods to `signCommitment`, `signHtlc`, `signRevoked`?

**Comment by t-bast** [comment]
It's a bit more complicated than that: the metric tags will be slightly inaccurate because some 2nd-level txs (when closing) will be incorrectly assigned to htlc / commit txs.

But the rate of closing txs will be negligible compared to the rate of signing updated commitment txs, so I think the metric can ignore this subtlety, but renaming the methods could be confusing for readers when studying the closing flows.

File: eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala [hunk 1]
```diff
@@ -97,12 +99,15 @@ class LocalChannelKeyManager(seed: ByteVector, chainHash: ByteVector32) extends
...
    * @param txOwner          owner of the transaction (local/remote)
    * @param commitmentFormat format of the commitment tx
-   * @return a signature generated with the private key that matches the input
-   *         extended public key
+   * @return a signature generated with the private key that matches the input extended public key
    */
   override def sign(tx: TransactionWithInputInfo, publicKey: ExtendedPublicKey, txOwner: TxOwner, commitmentFormat: CommitmentFormat): ByteVector64 = {
-    val privateKey = privateKeys.get(publicKey.path)
-    Transactions.sign(tx, privateKey.privateKey, txOwner, commitmentFormat)
+    val tags = TagSet.Empty.withTag(Tags.TxOwner, txOwner.toString).withTag(Tags.TxType, Tags.TxTypes.CommitTx)
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala:105]:
```suggestion
    // NB: not all those transactions are actually commit txes (especially during closing), but this is good enough for monitoring purposes
    val tags = TagSet.Empty.withTag(Tags.TxOwner, txOwner.toString).withTag(Tags.TxType, Tags.TxTypes.CommitTx)
```

File: eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala [hunk 2]
```diff
@@ -113,13 +118,16 @@ class LocalChannelKeyManager(seed: ByteVector, chainHash: ByteVector32) extends
...
    * @param commitmentFormat format of the commitment tx
-   * @return a signature generated with a private key generated from the input keys's matching
-   *         private key and the remote point.
+   * @return a signature generated with a private key generated from the input key's matching private key and the remote point.
    */
   override def sign(tx: TransactionWithInputInfo, publicKey: ExtendedPublicKey, remotePoint: PublicKey, txOwner: TxOwner, commitmentFormat: CommitmentFormat): ByteVector64 = {
-    val privateKey = privateKeys.get(publicKey.path)
-    val currentKey = Generators.derivePrivKey(privateKey.privateKey, remotePoint)
-    Transactions.sign(tx, currentKey, txOwner, commitmentFormat)
+    val tags = TagSet.Empty.withTag(Tags.TxOwner, txOwner.toString).withTag(Tags.TxType, Tags.TxTypes.HtlcTx)
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala:124]:
```suggestion
    // NB: not all those transactions are actually htlc txes (especially during closing), but this is good enough for monitoring purposes
    val tags = TagSet.Empty.withTag(Tags.TxOwner, txOwner.toString).withTag(Tags.TxType, Tags.TxTypes.HtlcTx)
```

Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala:126]: Done in https://github.com/ACINQ/eclair/pull/1659/commits/11e54785be3e67dc6c2598dbe60c9b45191c3216

File: eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala (hunk 1 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala:106]: Done in https://github.com/ACINQ/eclair/pull/1659/commits/11e54785be3e67dc6c2598dbe60c9b45191c3216

**Review by pm47 — APPROVED** [review]

**Pull request merged**